app/src/components/market/sections/market_buy/market_buy.tsx

```diff
@@ -73,12 +77,30 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
   const [tweet, setTweet] = useState('')
   const [newShares, setNewShares] = useState<Maybe<BigNumber[]>>(null)
 
+  useEffect(() => {
+    dispatch(fetchAccountBalance(account, provider, collateral))
+  }, [dispatch, account, provider, collateral])
+
+  const [collateralBalance, setCollateralBalance] = useState<BigNumber>(Zero)
+  const [collateralBalanceFormatted, setCollateralBalanceFormatted] = useState<string>(
+    formatBigNumber(collateralBalance, collateral.decimals),
+  )
+  const maybeCollateralBalance = useCollateralBalance(collateral, context)
+
```

Comment:
Why fetch balance from the reducer when we already have it in the marketMakerData object?

Reply:
@KadenZipfel 
Currently, currency selector is disabled.
But later, if users select another currency, I should get balance of the selected currency and show.

Reply:
Let me change back to use the balance in the marketMakerData